infrastructure-kotlin/src/main/kotlin/kotlin/KotlinLibraryPlugin.kt

```diff
@@ -24,18 +23,19 @@ public class KotlinLibraryPlugin : InfrastructurePlugin() {
         apply(plugin = "kotlin")
         val configPlugin = plugins.apply(KotlinConfigPlugin::class)
 
-        java {
-            targetCompatibility = JavaVersion.VERSION_1_8
-            sourceCompatibility = JavaVersion.VERSION_1_8
-        }
-
         // Enable Explicit API mode for libraries by default
         kotlin.explicitApi()
 
-        val extension = redmadrobotExtension
-        configureKotlin(extension.jvmTarget)
+        configureKotlin(configPlugin.jvmTarget)
         configureKotlinTest(configPlugin.testOptions)
         configureRepositories()
+
+        afterEvaluate {
```

Comment:
Почему тут afterEvaluate?

Reply:
This code will be applied before a user changed `redmadrobot.jvmTarget`. The order will be:
1. Apply KotlinLibraryPlugin (use `jvmTarget` and finalize it)
2. Change `redmadrobot.jvmTarget` (throws error because `jvmTarget` already finalized)

The only way to change `jvmTarget` without `afterEvaluate` - declare it in parent project so it will be configured before the plugin apply.